### ISSUE TITLE
PDF and ePub versions of the official documentation is required

### ISSUE THREAD

**Issue by camden-kid** [angular/angular#19459]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  
[ ] Feature request
[x] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
</code></pre>

## Current behavior

PDF and ePub versions of the official documentation are not available.

## Expected behavior

Would be good to have PDF and ePub versions of the official documentation to download.

## What is the motivation / use case for changing the behavior?

The documentation is large and not very easy to navigate. The crucial ability to add notes and bookmarks is very much needed.

## Environment

<pre><code>
Angular version: All

Browser:
- [ ] Chrome (desktop) version XX
- [ ] Chrome (Android) version XX
- [ ] Chrome (iOS) version XX
- [ ] Firefox version XX
- [ ] Safari (desktop) version XX
- [ ] Safari (iOS) version XX
- [ ] IE version XX
- [ ] Edge version XX
 
For Tooling issues:
- Node version: XX  
- Platform:  

Others:

</code></pre>

**Comment by lazarljubenovic** [comment]
I find the use-case very strange.

- It's without a doubt easier to searach a web-site which has a custom search functionality which is fully aware of what you'researching for and might improve in futue (to allow typos etc), while searching through a PDF document is a simple text-search.
- Adding bookmarks is quite possible because all titles and subtitles have anchor links, so you can reference a portion of text quite easy.
- I'm sure there are a lot of plugins which make it possible to stick a note to a page.

The only case where PDF might be prefereable is offline reading, but now we have service workers and PWAs for that. 

Not saying that this should _not_ be implemented, just stating that fom my POV I fail to see how this would be a priority for a long time.

**Comment by jenniferfell** [comment]
If I understand correctly..and just to confirm...your primary requests are to able to have personal (not public) annotations and bookmarks on the Angular docs. Is that correct? For bookmarks, you'd like an easy way to come back to something after you've found it. For annotations, you want to be able to write in the virtual margins. And it sounds like ideally this would be at the paragraph level, not just at the page level. Is that correct? Are your annotations mostly, "Here's the info about animations," (bookmark-like tags), or "Angular does x this way" (technical tips and ah-ha moments), or other types of notes? 

You're not trying to use the docs in an offline mode or print them out. Correct?

**Comment by petebacondarwin** [comment]
I don't believe we plan to render to non-HTML formats in the near future. @lazarljubenovic points out many of the reasons for requiring this can be achieve through better HTML presentation.
@camden-kid if you a specific need then (e.g. be able to place personal highlights and notes over the docs) then please open specific issues for those requirements and we can look into how best to solve them.

**Comment by camden-kid** [comment]
@jenniferfell @lazarljubenovic @petebacondarwin 

Sorry for the lack of detail in the original message. Jennifer, your first two paragraphs say perfectly what I would like to see.

Here are the things I would like to be able to do with the documentation:

- Mark a point in a page which I last read 
- Highlight text and add a note to that highlight
- Easily view all notes
- Bookmark sections of documentation or if it was an ePub bookmark a page
- Easily view all bookmarks
- View the documentation off line

This page is a perfect example of what I would use the above for - https://angular.io/guide/reactive-forms. Once I've added my highlights, notes, etc. I would like to be able to see them on their own page/section so that I can see what I've read and where I have been in the documentation.

It would be nice to highlight this for example:

`You don't create Angular form control objects. Angular directives create them for you, using the information in your data bindings. You don't push and pull data values. Angular handles that for you with ngModel. Angular updates the mutable data model with user changes as they happen.`

I often think that PDF/ePub is easier on the eye, especially when reading on a smart phone. One can also change the font, colours, etc.

**Comment by mgatto** [comment]
Reading the docs on an e-ink reader device would be great.

**Comment by mlc-mlapis** [comment]
@mgatto ... Angular documentation is changing relatively a lot so it would mean that there will be a new PDF each day maybe several times a day. Do you still think that it is the advantage to have it?

**Comment by jenniferfell** [comment]
@camden-kid Thanks for the additional detail above. (Sorry for this very delayed thanks. I think I have my notifications set correctly now. Oops.)

**Comment by seinecle** [comment]
I bring my support for this issue: reading in print is useful to some of us - I'd say it is a matter of taste / reading habits: some just prefer ebooks to web pages, and some of use like to print (my case): just look at the number of books on Angular sold as hard copies on Amazon.

The code changes a lot, so a toolchain that would generate a pdf at regular intervals (once per week) would be great.

**Comment by petebacondarwin** [comment]
> The only case where PDF might be prefereable is offline reading, but now we have service workers and PWAs for that.

The idea behind this comment was that through a Service Worker, we can provide the angular.io docs website to be read offline.

**Comment by jenniferfell** [comment]
Related issue about just having the cheat sheet offline: https://github.com/angular/angular/issues/24555

**Comment by jenniferfell** [comment]
Related issue about offline use of doc in general: https://github.com/angular/angular/issues/22580

**Comment by petebacondarwin** [comment]
Re-opening as this issue still has some interest. See #41558.
@aikidave - do you have a view on this?

**Comment by PingPongSet** [comment]
@petebacondarwin thanks for handling it.

One reason is that it tells me how long it will take to learn each section based on a PDF.

It is also easier to use on other scenarios, for example, offline, reading it on papper.

**Comment by jorgeriv** [comment]
I second this. In my case is not because the off-line which can be achieved with the service worker, I would like to be able to read in a e-ink device since reading in the computer screen is very hard on my eyes.

**Comment by aikidave** [comment]
Thanks for the feedback on this issue, all! I will look into what it might take to create PDF/ePub versions of the Angular documentation!

**Comment by PingPongSet** [comment]
@aikidave 

Thanks. One option, if it is easier, is to combine all htmls or main htmls into one html, users can use browser's print to generate a PDF.

like "Print entire section" on the top right from the link below.
https://kubernetes.io/docs/concepts/_print/

**Comment by petebacondarwin** [comment]
> Thanks. One option, if it is easier, is to combine all htmls or main htmls into one html, users can use browser's print to generate a PDF.

That would be quite possible to do with our current setup.

**Comment by PingPongSet** [comment]
One issue with PDF is that pages that use JS will not work, like this below

https://angular.io/guide/dynamic-component-loader#the-adcomponent-interface

That's why I also requested HTML for download, if possible.
